### PR TITLE
Fallback implementation of sizehint! for AbstractDict

### LINKED ISSUE #44686: Plots.jl not working on Julia master -- LoadError: MethodError: no method matching sizehint!(::RecipesPipeline.DefaultsDict, ::Int64)
Not sure if this is a Julia issue or a package issue but Plots.jl is currently broken on the Julia master. I filed an issue in Plots.jl and referencing here in case the issue lies with changes in Julia master:

https://github.com/JuliaPlots/Plots.jl/issues/4144#issue-1174595889

versioninfo:

```
Julia Version 1.9.0-DEV.218
Commit 1b686b2633 (2022-03-19 09:28 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin21.4.0)
  CPU: 8 × Apple M1
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, apple-m1)
  Threads: 1 on 4 virtual cores
```

### PR DESCRIPTION
**Pull request by petvana**
Fixes #44686 + additional `haslength` check + docs that `sizehint!` returns the collection.

### CODE REVIEW

File: base/abstractdict.jl [hunk 1]
```diff
@@ -522,6 +522,9 @@ function ==(l::AbstractDict, r::AbstractDict)
     return anymissing ? missing : true
 end
 
+# Fallback implementation
+sizehint!(s::AbstractDict, n) = nothing
```
Comment (fredrikekre) [inline base/abstractdict.jl:526]:
```suggestion
sizehint!(s::AbstractDict, n) = s
```

**Review by fredrikekre — COMMENTED** [review]
Perhaps add a small test?

File: base/abstractdict.jl (hunk 1 above)
Comment (petvana) [reply base/abstractdict.jl:526]: If so, should it be changed also for `AbstractSet`?
https://github.com/JuliaLang/julia/blob/1b686b2633052e21760cc936abe3a54834927948/base/abstractset.jl#L4

Comment (fredrikekre) [reply base/abstractdict.jl:526]: Hmm. Yea, I would expect the collection to be returned like most APIs like this.

**Comment by petvana** [comment]
> Perhaps add a small test?

Added for both `AbstractSet` and `AbstractDict`.

**Pull request merged**